twoliter/src/cmd/build.rs

```diff
@@ -25,10 +33,94 @@ pub(crate) struct BuildVariant {
     /// The architecture to build for.
     #[clap(long = "arch", default_value = "x86_64")]
     arch: String,
+
+    /// The variant to build.
+    variant: String,
 }
 
 impl BuildVariant {
     pub(super) async fn run(&self) -> Result<()> {
-        Ok(())
+        let project = project::load_or_find_project(self.project_path.clone()).await?;
+        let token = project.token();
+        let tempdir = tools_tempdir()?;
+        install_tools(&tempdir).await?;
+        let makefile_path = tempdir.path().join("Makefile.toml");
+        let packages_dir =
+            TempDir::new().context("Unable to create a tempdir for Twoliter's packages")?;
+
+        let sdk_container = DockerContainer::new(
+            format!("sdk-{}", token),
+            project
+                .sdk(&self.arch)
+                .context("The project was missing an sdk")?
+                .uri(),
+        )
+        .await?;
+        sdk_container
+            .cp(Path::new("twoliter/alpha/build/rpms"), packages_dir.path())
+            .await?;
+
+        let rpms_dir = project.project_dir().join("build").join("rpms");
+        fs::create_dir_all(&rpms_dir)?;
+        debug!("Moving rpms to build dir");
+        for maybe_file in fs::read_dir(packages_dir.path().join("rpms"))? {
+            let file = maybe_file?;
+            debug!("Moving '{}'", file.path().display());
+            fs::rename(file.path(), rpms_dir.join(file.file_name()))?;
+        }
+
+        let mut created_files = Vec::new();
+
+        let in_tree_sbkeys_dir = project.project_dir().join("sbkeys");
+        let sbkeys_dir: Box<dyn AsRef<Path>> = if in_tree_sbkeys_dir.is_dir() {
+            Box::new(in_tree_sbkeys_dir)
+        } else {
+            // Create a temp sbkeys directory
+            debug!("sbkeys dir not found. Creating a temporary directory");
+            let sbkeys_dir = TempDir::new().context("Unable to create a tempdir for sbkeys")?;
+            trace!("Using sbkeys tempdir: '{}'", sbkeys_dir.path().display());
+            sdk_container
+                .cp(Path::new("twoliter/alpha/sbkeys"), sbkeys_dir.path())
+                .await?;
+            Box::new(sbkeys_dir)
+        };
+
+        // TODO: Remove once models is no longer conditionally compiled.
+        // Create the models directory for the sdk to mount
+        let models_dir = project.project_dir().join("sources/models");
+        if !models_dir.is_dir() {
+            debug!("models source dir not found. Creating a temporary directory");
+            fs::create_dir_all(&models_dir.join("src/variant"))
+                .context("Unable to create models source directory")?;
+            created_files.push(models_dir)
+        }
```

Comment:
Can we `mkdir -p` this inside the embedded `Makefile.toml` so that the directory only gets created in the build context, not out on the host? That would eliminate the need for the created files cleanup later.